tqdm.py

```diff
@@ -0,0 +1,63 @@
+import collections
+import time
+import copy
+
+
+class Tqdm(collections.Iterator):
+    """
+    Tqdm class will receive a given iterator and will wrap and analyze it's execution with aditional metadata such as
+    running time and progress status
+    """
+
+    def __init__(self, func, length=None):
+        self.func = func
+        self.index = 0
+        self.iter_length = length
+
+    def __iter__(self):
+        if hasattr(self.func, '__len__'):
+            self.iter_length = len(self.func)
+
+        self.iter = iter(self.func)
+        self.start_time = time.time()
+        return self
+
+    def __next__(self):
+        next_val = next(self.iter)
+
+        self.index += 1
+        output = ''
+        output += f'Running iteration number {self.index}, '
```

Comment:
No need to save `output` and update it, you can use simple prints with `end=''`

Reply:
I did before and it looks worse.... which is why I switched to output accumulation... also better for logging or any other output data piping